lean-repo-utils.php

```diff
@@ -211,7 +214,7 @@ function push_back($fullRepository, $workDir, $upstreamRepoWithCredentials, $bui
         print "FAILED with $status\n";
     }
     // We don't want to commit the build-metadata to the canonical repository.
-    passthru("git --git-dir=$canonicalRepository/.git -C $fullRepository reset HEAD $buildMetadataFile");
+    passthru("git --git-dir=$canonicalRepository/.git -C $fullRepository reset HEAD $buildMetadataFile dev-master");
```

Comment:
Is `dev-master` always correct? Maybe we need to look up the default branch name, in case it has been changed.

Reply:
this is a weird file that was being added to the external VCS, here I'm just removing it. To be honest, I'm not 100% sure why it exists in first place but I just removed it here to avoid it getting pushed back.

And it was dev-master in my tests with drupal9

Reply:
I was concerned that some customers might have changed it (e.g. in custom templates), but for build tools we still need to maintain a strong git repo : pantheon repo branch mapping, so the default branch has to remain "master" here. So this assumption is safe.